portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +31,101 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  int numberOfCommentsToDisplay = 5;
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Deus ");
+    System.out.println("INSIDE THE DO GET FUNCTION");
+    System.out.println(numberOfCommentsToDisplay);
+    Query query = new Query("entityComment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String name = (String) entity.getProperty("name");
+      String commentText = (String) entity.getProperty("comment");
+      long timestamp = (long) entity.getProperty("timestamp");
+      System.out.println(id + "\t" + name + "\t" +  commentText + "\t" + timestamp);
+
+      Comment comment = new Comment(id, name, commentText, timestamp);
+      comments.add(comment);
+    }
+
+    response.setContentType("application/json");
+    // Convert the server stats to JSON
+    String json = "";
+    if (comments.size() > 0) {
+        json = convertToJson(comments.subList(0, numberOfCommentsToDisplay));
+    }
+    
+    // Send the JSON as the response
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    System.out.println("INSIDE THE DO POST FUNCTION");
+    numberOfCommentsToDisplay = getUserChoice(request);
+    System.out.println("USER CHOICE IS: " + numberOfCommentsToDisplay);
+    String name = request.getParameter("name-input");
+    String newComment = request.getParameter("comment-input");
+    long timestamp = System.currentTimeMillis();
+
+    if (newComment != null){
+      Entity commentEntity = new Entity("entityComment");
+      commentEntity.setProperty("name", name);
+      commentEntity.setProperty("comment", newComment);
+      commentEntity.setProperty("timestamp", timestamp);
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(commentEntity);
+    }
+
+    
+    // Redirect back to the HTML page.
+    response.sendRedirect("index.html");
+  }
+
+  /**S
+   * Converts a ServerStats instance into a JSON string using the Gson library. Note: We first added
+   * the Gson library dependency to pom.xml.
+   */
+  private String convertToJson(List<Comment> comments) {
```

Comment:
Rather than generate a JSON string by hand consider using the Gson library to do so. This will allow you to easily convert javascript objects to JSON and vice-versa without having to worry about edge cases (such as escape characters in your text).